### PR TITLE
LPC17_40 CAN driver SocketCAN enforce TX fifo behaviour

### PR DESCRIPTION
**Pull request by PetervdPerk**
## Summary
The SocketCAN driver expects a FIFO behaviour yet LPC17_40 didn't enforce this.
This changes enables prioritization on transmit function to enforce this.

## Impact
Fix transmit behaviour

## Testing
Tested on a LPC1768

### CODE REVIEW

File: arch/arm/src/lpc17xx_40xx/lpc17_40_can.c [hunk 1]
```diff
@@ -1049,17 +1062,22 @@ static int lpc17can_txpoll(struct net_driver_s *dev)
 
   if (priv->dev.d_len > 0)
     {
-      /* Send the packet */
-
-      lpc17can_transmit(priv);
+      if (!devif_loopback(&priv->dev))
```
Comment (xiaoxiang781216) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_can.c:1065]: devif_loopback is specific for ipv4 and ipv6 protocol, is it good to call it in can driver?

Comment (PetervdPerk) [reply arch/arm/src/lpc17xx_40xx/lpc17_40_can.c]: Thanks for pointing that out, got mixed with backporting this from my older branch.
Should be okay now.

**Review by davids5 — APPROVED** [review]
LGTM

File: arch/arm/src/lpc17xx_40xx/lpc17_40_can.c [hunk 2]
```diff
@@ -1474,6 +1505,13 @@ static void can_interrupt(struct lpc17_40_can_s *dev)
       can_txdone(dev);
 #elif defined(SOCKET_CAN)
       NETDEV_TXDONE(&priv->dev);
+      regval = can_getreg(priv, LPC17_40_CAN_GSR_OFFSET);
+      if (regval & CAN_GSR_TBS)
```
Comment (pkarashchenko) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_can.c:1509]: Optional
```suggestion
      if ((regval & CAN_GSR_TBS) != 0)
```

File: arch/arm/src/lpc17xx_40xx/lpc17_40_can.c [hunk 3]
```diff
@@ -1455,6 +1479,13 @@ static void can_interrupt(struct lpc17_40_can_s *dev)
...
```
Comment (pkarashchenko) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_can.c:1483]: Optional
```suggestion
      if ((regval & CAN_GSR_TBS) != 0)
```

File: arch/arm/src/lpc17xx_40xx/lpc17_40_can.c [hunk 4]
```diff
@@ -1436,6 +1453,13 @@ static void can_interrupt(struct lpc17_40_can_s *dev)
...
```
Comment (pkarashchenko) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_can.c:1457]: Optional:
```suggestion
      if ((regval & CAN_GSR_TBS) != 0)
```

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by davids5 — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Could you please squash commits?

**Comment by acassis** [comment]
> Could you please squash commits?

And now we have a documentation to explain how to do it:

https://nuttx.apache.org/docs/latest/contributing/making-changes.html#how-to-include-the-suggestions-on-your-pull-request

Of course, @PetervdPerk is not a New Kind on the Block, the idea was to help new contributors

**Comment by xiaoxiang781216** [comment]
Let' ignore the macOS ci broken.

**Pull request merged**